src/object/srv_obj.c

```diff
@@ -2316,7 +2319,27 @@ obj_verify_bio_csum(crt_rpc_t *rpc, daos_iod_t *iods,
 		daos_sgl_fini(&sgl, false);
 
 		if (rc != 0) {
-			D_ERROR("Verify failed: %d\n", rc);
+			if (iod->iod_type == DAOS_IOD_SINGLE) {
+				D_ERROR("Data Verification failed (object: "
+				DF_OID" dkey: %s, akey: %s): %d\n",
+					DP_OID(orw->orw_oid.id_pub),
+					(char *)orw->orw_dkey.iov_buf,
+					(char *)iod->iod_name.iov_buf, rc);
+			}
+			if (iod->iod_type == DAOS_IOD_ARRAY) {
+				D_ERROR("Data Verification failed (object: "
+				DF_OID" dkey: %s, akey: %s, "
+				"extent: [%lu-%lu]): %d\n",
```

Comment:
same comment as above.   You might also add a DF_RECX somewhere.

Reply:
Created a DF_RECX and using it throughout